torch/nn/modules/module.py

```diff
@@ -101,12 +101,15 @@ def register_buffer(self, name, tensor):
             >>> self.register_buffer('running_mean', torch.zeros(num_features))
 
         """
-        if hasattr(self, name) and name not in self._buffers:
-            raise KeyError("attribute '{}' already exists".format(name))
+        if name is None or not isinstance(name, torch._six.string_classes):
```

Comment:
This `is None` check seems unnecessary. The `isinstance` will catch it anyways. Can you remove it?